lib/mix/lib/mix/compilers/elixir.ex

```diff
@@ -377,6 +377,24 @@ defmodule Mix.Compilers.Elixir do
     end
   end
 
+  defp warning_diagnostics(sources) do
+    Enum.flat_map(sources, fn source(source: source, warnings: warnings) ->
+      for {line, message} <- warnings do
+        diagnostic({source, line, message}, :warning)
+      end
+    end)
+  end
+
+  defp diagnostic({file, line, message}, severity) do
+    %Mix.Task.Compiler.Diagnostic{
+      file: Path.absname(file),
+      position: line,
```

Comment:
Should we call this "line" since we may want to add a column field later on too?

Reply:
Nevermind, I read the docs later on. :)